### DISCUSSION TITLE
Issue with initial button press 504 gateway time-out

### DISCUSSION THREAD

**Discussion by cloudwales** [livewire/livewire#4038] [1. Help]
Hi. I'm getting an odd issue where I have a button that you click and it imports csv files from S3 storage and imports the contents into the database. 

I have a wire:poll.3000ms running on the page, this checks for new files on S3 and displays them on the list to import. This works fine. When I click the Import button which is `wire:click="importFiles"` something interrupts the polling, the network tab in developer tools shows it going to (pending). It then throws an error 504 gateway time-out, but carries on working as it should, but while its working ok in the backend, it throws the error modal up. 

Any ideas?

Livewire Component::
```
class CsvImport extends Component
{
    public function render()
    {
        $answer_imports = AnswerImport::all();
        $imported_files = [];
        foreach ($answer_imports as $import) {
            $imported_files[] = $import->file_name;
        }
        $files = collect(Storage::disk('do_spaces')->files('a_location'));

        return view('livewire.answers.csv-import')->with([
            'files' => $files,
            'imported_files' => $imported_files
        ]);
    }


    public function importFiles()
    {
        $answers = collect(Storage::disk('do_spaces')->allFiles('a_location'));
        
            foreach ($answers as $file) {
                $answer_file = AnswerImport::where('file_name', $file)->first();
                if (!$answer_file) {
                    Excel::import(new AnswersImport, $file, 'do_spaces');
                    AnswerImport::create([
                    'file_name' => $file
                ]);
            }
            
        }
    }
}
```

Blade button::
```
<button wire:click="importFiles"
                    class="inline-flex items-center px-4 py-2 text-sm font-medium leading-5 text-white transition duration-150 ease-in-out bg-indigo-600 border border-transparent rounded-md hover:bg-indigo-500 focus:outline-none focus:shadow-outline-indigo focus:border-indigo-700 active:bg-indigo-700">
     Import
</button>
```

**Comment by joshhanley** [comment]
@cloudwales maybe it's actually running ok, but taking too long for the browser or maybe the webserver. You'll have to research to see if there is a way to extend the timeout length.

But I would actually fire that off to a queue job which goes and checks S3 and does whatever it needs to do. You can then just setup your `wire:poll` which will refresh the list of available files everytime it polls (but not deal with the finding/ handling of new ones).

Hope this helps!

**Comment by cloudwales** [comment]
Hi @joshhanley, Thanks for the reply. Thats what I done eventually was set it as a job and let it run in the background and have `wire:poll` update the frontend. It was strange, the initial request timed out, but it was running in the background... Creating a job for this worked for me. 

Many thnaks

**Reply by joshhanley** [reply]
No worries! 🙂 yeah probably hit nginx/ apache's timeout limit.

**Comment by ekvedaras** [comment]
I have kind of similar (maybe) issue with file uploader (in filament). Occasionally the request to getUploadedFiles just hangs until we get 504. In php-fpm logs I see that child process exited with error code 124. All other requests work except that one (even while that one is hanging, but thats I guess because they land on a different child process). After it times out and child process gets restarted, the livewire call works fine (pretty much instant). This is related also to S3. I would guess something about AWS calls just remains dangling and blocks further requests. 
The annoying part is that in Laravel Nova where we use some other 3rd party plugin for spatie media library it does not have this problem 🤷 even though they do get those files in a very similar manner. 

Update: also saw now that the hanging child process is eating CPU. So must be looping somewhere endlessly
